typescript/hx-adl/src/gen-sqlschema.ts

```diff
@@ -10,45 +10,59 @@ import { snakeCase } from "change-case";
-   .option('--postgres', 'Generate sql for postgres')
-   .option('--postgres-v2', 'Generate sql for postgres (model version 2)')
-   .option('--mssql', 'Generate sql for microsoft sqlserver')
-   .option('--extension <ext>', 'Add to included sql extensions', collect, [])
-   .description('Generate a db schema from ADL files')
-   .action( (adlFiles:string[], cmd:{}) => {
-     const adlSearchPath: string[] = cmd['searchdir'];
-     const extensions: string[] = cmd['extension'];
-     const templates: Template[] = parseTemplates(cmd['outtemplatesql'] || []);
-
-     let outfile: string = cmd['outfile'];
-     if (cmd['outputdir']) {
-       outfile = cmd['outputdir'] + '/create.sql';
-     }
-
-     let outmetadata: string | null = cmd['outmetadata'] || null;
-
-     let dbProfile = postgresDbProfile;
-     if (cmd['postgresV2']) {
-       dbProfile = postgres2DbProfile;
-     }
-     if (cmd['mssql']) {
-       dbProfile = mssql2DbProfile;
-     }
-
-     generateSql({adlFiles, adlSearchPath, outfile, outmetadata, extensions, templates, dbProfile});
-   });
+    .command("sql [adlFiles...]")
+    .option('-I, --searchdir <path>', 'Add to adl searchpath', collect, [])
+    .option('--outfile <path>', 'the resulting sql file', 'create.sql')
+    .option('--outputdir <dir>', 'the directory into which the sql is written (deprecated)')
+    .option('--outmetadata <path>', 'sql to insert the model metadata')
+    .option('--outtemplatesql <paths>', 'generate extra sql from a mustache template', collect, [])
+    .option('--postgres', 'Generate sql for postgres')
+    .option('--postgres-v2', 'Generate sql for postgres (model version 2)')
+    .option('--mssql', 'Generate sql for microsoft sqlserver')
+    .option('--extension <ext>', 'Add to included sql extensions', collect, [])
+    .option('--as-alter-statements', 'Generate sql in alter format', false)
+    .description('Generate a db schema from ADL files')
+    .action((adlFiles: string[], cmd: {}) => {
+      const adlSearchPath: string[] = cmd['searchdir'];
+      const extensions: string[] = cmd['extension'];
+      const templates: Template[] = parseTemplates(cmd['outtemplatesql'] || []);
+
+      let outfile: string = cmd['outfile'];
+      if (cmd['outputdir']) {
+        outfile = cmd['outputdir'] + '/create.sql';
+      }
+
+      let outfileCreate: string = "01-create.sql"
+      let outfileRearrange: string = "02-rearrange.sql"
+      let outfileConstrant: string = "03-constraints.sql"
+      let outfileDrop: string = "04-drop.sql"
+      if (cmd['outputdir']) {
+        outfileCreate = cmd['outputdir'] + outfileCreate;
+        outfileRearrange = cmd['outputdir'] + outfileRearrange;
+        outfileConstrant = cmd['outputdir'] + outfileConstrant;
+        outfileDrop = cmd['outputdir'] + outfileDrop;
+      }
+      const outfiles: string[] = [outfileCreate, outfileRearrange, outfileConstrant, outfileDrop];
```

Comment:
It's a bit strange that there is an array of `outfiles` and entry zero is the `create.sql` and then the code never refers to it.